front/src/app/components/forms/commons/AgencySelector.tsx

```diff
@@ -277,3 +234,20 @@ const getAgencyPlaceholder = (
   if (numberOfAgencies === 0) return "Aucune agence dans ce département";
   return "Veuillez sélectionner une structure";
 };
+
+const agencyOptionsInSelectorFromAgencies = (
+  agencies: AgencyOption[],
+  agencyKind: AgencyKindForSelector,
+) =>
+  agencies
+    .filter(({ kind }) => (agencyKind === "all" ? true : kind === agencyKind))
```

Comment:
c'est peut-être plus simple comme ça :
```suggestion
    .filter(({ kind }) => (agencyKind === "all" || kind === agencyKind))
```